### PR TITLE
Revert PCP-788, merge 1.7.x, restore PCP-771

### PR DESCRIPTION
**Pull request by MikaelSmith**
Reverts PCP-788, then does a merge-up from 1.7.x, and restores PCP-771.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,6 @@
 This is a feature release.
 
 * [PCP-786](https://tickets.puppetlabs.com/browse/PCP-786) Send error response to status if found
```
Comment (ekinanp) [inline CHANGELOG.md]: I'm guessing that PCP-771 will be added to the CHANGELOG once we figure out the solaris failures?

Comment (MikaelSmith) [reply CHANGELOG.md]: Yeah, we'll put it under the appropriate version.

**Comment by MikaelSmith** [comment]
This should be safe to merge after https://github.com/puppetlabs/puppet-agent/pull/1164 is in.

**Comment by MikaelSmith** [comment]
Actually, no reason to hold off merging it. Tests are already failing.

**Pull request merged**